scipy/special/generate_ufuncs.py

```diff
@@ -393,13 +395,247 @@ def errprint(inflag=None):
         return sf_error.get_print()
 """
 
-EXTRA_CODE_BOTTOM = """\
+UFUNCS_EXTRA_CODE_BOTTOM = """\
 #
 # Aliases
 #
 jn = jv
 """
 
+CYTHON_SPECIAL_PXD = """\
+# This file is automatically generated by generate_ufuncs.py.
+# Do not edit manually!
+"""
+
+CYTHON_SPECIAL_PYX = """\
+# This file is automatically generated by generate_ufuncs.py.
+# Do not edit manually!
+\"\"\"
+.. highlight:: cython
+
+================================
+Cython API for Special Functions
+================================
+
+Scalar, typed versions of many of the functions in ``scipy.special``
+can be accessed directly from Cython; the complete list is given
+below. Functions are overloaded using Cython fused types so their
+names match their ufunc counterpart. The module follows the following
+conventions:
+
+- If a function's ufunc counterpart returns multiple values, then the
+  function returns its outputs via pointers in the final arguments
+- If a function's ufunc counterpart returns a single value, then the
+  function's output is returned directly.
+
+The module is usable from Cython via::
+
+    cimport scipy.special.cython_special
+
+Error Handling
+==============
+
+Error handling works the same as in `scipy.special`; in particular
+error messages can be controlled by ``cython_special.errprint``.
+
+Available Functions
+===================
+
+FUNCLIST
+\"\"\"
+cimport numpy as np
+from numpy cimport (
+    npy_float, npy_double, npy_longdouble,
+    npy_cfloat, npy_cdouble, npy_clongdouble,
+    npy_int, npy_long,
+    NPY_FLOAT, NPY_DOUBLE, NPY_LONGDOUBLE,
+    NPY_CFLOAT, NPY_CDOUBLE, NPY_CLONGDOUBLE,
+    NPY_INT, NPY_LONG)
+
+cdef extern from "numpy/ufuncobject.h":
+    int PyUFunc_getfperr() nogil
+
+cdef public int wrap_PyUFunc_getfperr() nogil:
+    \"\"\"
+    Call PyUFunc_getfperr in a context where PyUFunc_API array is initialized;
+    this avoids messing with the UNIQUE_SYMBOL #defines
+    \"\"\"
+    return PyUFunc_getfperr()
+
+cdef extern from "numpy/npy_math.h":
+    double NPY_NAN
+
+cimport sf_error
+cimport _complexstuff
+cimport scipy.special._ufuncs_cxx
+from . import _ufuncs
+
+ctypedef long double long_double
+ctypedef float complex float_complex
+ctypedef double complex double_complex
+ctypedef long double complex long_double_complex
+
+cdef class _float:
```

Comment:
I think in the tests you will have `type(cython_special._long_double(x).val) is float` iow. this probably doesn't help in choosing the correct signatures.